WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingBottomSheetFragment.kt

```diff
@@ -50,30 +61,73 @@ class PrepublishingBottomSheetFragment : WPBottomSheetDialogFragment() {
         }
     }
 
-    private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory)
+    private fun initViewModels() {
+        prepublishingViewModel = ViewModelProviders.of(this, viewModelFactory)
                 .get(PrepublishingViewModel::class.java)
+
+        prepublishingActionsViewModel = ViewModelProviders.of(this, viewModelFactory)
+                .get(PrepublishingActionsViewModel::class.java)
+
+        prepublishingActionsViewModel.prepublishingActionType.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let { actionType ->
+                prepublishingViewModel.updateCurrentActionTypeState(actionType)
+            }
+        })
+
+        prepublishingViewModel.currentActionsState.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let { actionState ->
+                navigateToScreen(actionState)
+            }
+        })
+
+        var actionsState = arguments?.getParcelable<ActionsState>(KEY_TAGS_ACTION_STATE)
+        if (actionsState == null) {
+            actionsState = ActionsState(tagsActionState = TagsActionState(tags = null))
+        }
```

Comment:
I'm not too sure about this approach. I know there must be a better way to do this. My thought process is I want to restore the state of the bottom sheet and if it doesn't exist I want to utilize an initial state. I think I might need to rename this as well.

Reply:
Again I don't have enough context yet to provide a better approach. However, I'd consider moving this into a VM as it's a logic.

Reply:
Yes, it is. Thank you!